### PR TITLE
Ccls

### PR DESCRIPTION
**Pull request by govi218**
Ref: #1 

Added ccls mode.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "rtags"]
+	path = rtags
+	url = https://github.com/Andersbakken/rtags.git
```
Comment (shaneikennedy) [inline .gitmodules]: I don't think we need this anymore

File: README.md [hunk 2]
```diff
@@ -18,3 +18,25 @@ $ brew cask install emacs
...
+
+``` sh
+# Start the RTags daemon (rdm)
+$ ./bin/rdm &
+
+# Index the RTags project, and wait until rdm is silent
+$ ./bin/rc -J .
+```
+
+For more information, refer to the [Rtags Github repo](https://github.com/Andersbakken/rtags).
```
Comment (shaneikennedy) [inline README.md]: or this

File: modules/my-cpp.el [hunk 3]
```diff
@@ -0,0 +1,18 @@
+;;; package --- Summary
+
+;;; Commentary:
+
+;;; Code:
+(use-package lsp-mode :commands lsp)
+(use-package lsp-ui :commands lsp-ui-mode)
+(use-package company-lsp :commands company-lsp)
```
Comment (shaneikennedy) [inline modules/my-cpp.el:8]: All of these already live in `my-editing`, so we can remove this 👍

File: modules/my-cpp.el [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+(use-package ccls
+  :hook ((c-mode c++-mode objc-mode cuda-mode) .
+         (lambda () (require 'ccls) (lsp))))
+(setq lsp-diagnostic-package nil)
+(setq-default flycheck-disabled-checkers '(c/c++-clang c/c++-cppcheck c/c++-gcc))
+(setq ccls-executable "/path/to/ccls/Release/ccls")
```
Comment (shaneikennedy) [inline modules/my-cpp.el]: 😅 Don't think this will work

**Review by shaneikennedy — CHANGES_REQUESTED** [review]
Needs a bit of updating but the lsp approach is much nicer :)

**Comment by shaneikennedy** [comment]
You could also squash these commits down into a single commit since the changes required is basically just adding my-cpp with the ccls package, no need to have rtags in the commit history  or the merge commits (use git pull -r origin/master for a cleaner merge) @govi218

File: modules/my-cpp.el (hunk 4 above)
Comment (govi218) [reply modules/my-cpp.el]: You mean `(setq lsp-diagnostic-package nil)`? I ripped that from the ccls github, originally `(setq lsp-prefer-flymake nil)`, but emacs said that was deprecated and favored this.

Comment (shaneikennedy) [reply modules/my-cpp.el]: `(setq ccls-executable "/path/to/ccls/Release/ccls")` I'm quite sure this path doesn't exists on either of our systems

**Pull request closed without merging**